docs/developer-docs/smart-contracts/advanced-features/async-code.mdx

```diff
@@ -1,90 +1,94 @@
 ---
-keywords: [advanced, concept, async code, inter-canister calls, async inter-canister, async]
+keywords: [beginner, concept, async code, inter-canister calls, async inter-canister, async]
 ---
 
 import { MarkdownChipRow } from "/src/components/Chip/MarkdownChipRow";
 
-# Async code and inter-canister calls
+# Inter-canister calls & async code
 
-<MarkdownChipRow labels={["Advanced", "Concept"]} />
+<MarkdownChipRow labels={["Beginner", "Concept"]} />
 
 ## Overview
 
-In programming, the [async/await pattern](https://en.wikipedia.org/wiki/Async/await) is a syntactic feature of many programming languages that allows an asynchronous, non-blocking function to be structured in a similar way to an ordinary synchronous function.
+The ICP allows canisters to seamlessly interact with other canisters by calling their methods, just like external users call canisters. You are likely to need inter-canister calls early on: they are necessary to transfer the ICP token, or to access certain system functionality through the [management canister](/docs/current/developer-docs/smart-contracts/advanced-features/management-canister). This remote procedure call (RPC) mechanism, based on a request-response paradigm, will feel familiar if you are coming from Ethereum or other smart-contract enabled blockchains, but there are also important differences that you should also understand:
 
-On ICP, `async/await` is useful in the context of inter-canister calls. Inter-canister calls cannot be handled synchronously (as the two canisters
-interacting might not even be in the same subnet), so when a canister sends a call to another, it will have to wait until a response comes back
-before it can continue processing the call it received.
+- On ICP, inter-canister calls are **asynchronous**. That means that while an inter-canister call is in progress other calls can still be processed. This significantly increases the canisters' scalability. However, it also means that canister authors have to correctly handle calls executing concurrently. In particular, this differs from the transactional nature of Ethereum's smart contract calls.
```

Comment:
Reworded to cover your first point.

You're also right that heartbeats can be executed concurrently, but I'd rather leave it out here since here this is meant to be an intro document (I wouldn't expect the users to know about heartbeats/timers yet)